torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -161,25 +119,49 @@ void ThreadPredicates::updateBitSet(Expr* expr) {
   for (const auto* out : expr->outputs()) {
     if (!ir_utils::isTV(out))
       continue;
-    thread_predicates[ir_utils::asConstTV(out)] = output_preds;
+    thread_predicates_[ir_utils::asConstTV(out)] = output_preds;
   }
 }
-ThreadPredicates::ThreadPredicates(Fusion* _fusion) : fusion_(_fusion) {
-  for (auto inp : fusion_->inputs())
-    if (ir_utils::isTV(inp))
-      thread_predicates[ir_utils::asConstTV(inp)] = std::bitset<num_p_type>();
-}
 
-std::unordered_map<const TensorView*, Bool*> ThreadPredicates::compute(
-    Fusion* fusion) {
-  ThreadPredicates tp(fusion);
-  for (auto expr : fusion->exprs(true))
-    tp.updateBitSet(expr);
-  std::unordered_map<const TensorView*, Bool*> preds;
-  for (auto entry : tp.thread_predicates) {
-    preds[entry.first] = getThreadPredicate(entry.second);
+ThreadPredicateMap::ThreadPredicateMap(Fusion* _fusion) : fusion_(_fusion) {
+  // Why is this needed?
```

Comment:
why is what needed? running through the inputs?

Reply:
Yes. Why populating the mapping for the inputs?

Reply:
I think it's because we go through exprs propagating their inputs to their outputs. We need initial values on inputs to start this propagation.

Reply:
Understood. Removed the comment.